bundle-size/app.js

```diff
@@ -208,47 +269,79 @@ module.exports = app => {
     };
 
     try {
-      const bundleSizesJson = JSON.parse(
+      const masterBundleSizes = JSON.parse(
         await getBuildArtifactsFile(github, `${baseSha}.json`)
       );
-      const baseBundleSize = bundleSizesJson['dist/v0.js'];
 
-      const bundleSizeDelta = bundleSize - baseBundleSize;
-      const bundleSizeDeltaFormatted = formatBundleSizeDelta(bundleSizeDelta);
+      // Calculate and collect all (non-zero) bundle size deltas and list all
+      // dist files that are missing either from master or from this pull
+      // request.
+      const otherBundleSizeDeltas = [];
+      const missingBundleSizes = [];
+      for (const [file, baseBundleSize] of Object.entries(masterBundleSizes)) {
+        if (file === 'dist/v0.js') {
+          // For now, skip dist/v0.js because it gets special handling.
+          continue;
+        }
+
+        if (!(file in prBundleSizes)) {
+          missingBundleSizes.push(`* ${file}: missing in pull request`);
+          continue;
+        }
+
+        const bundleSizeDelta = prBundleSizes[file] - baseBundleSize;
+        if (bundleSizeDelta !== 0) {
+          otherBundleSizeDeltas.push(
+            `* ${file}: ${formatBundleSizeDelta(bundleSizeDelta)}`
+          );
+        }
+      }
+
+      for (const [file, prBundleSize] of Object.entries(prBundleSizes)) {
+        if (!(file in masterBundleSizes)) {
+          missingBundleSizes.push(
+            `* ${file}: (${prBundleSize} KB) missing in \`master\``
+          );
+        }
+      }
+
+      if (otherBundleSizeDeltas.length === 0) {
+        otherBundleSizeDeltas.push(
+          '* No other bundle sizes reported in this pull request'
+        );
+      }
+
+      // TODO(#271, danielrozenberg): this is a transitionary solution. This API
+      // endpoint accepts a JSON with multiple bundle sizes, but for now it only
+      // blocks on size changes in dist/v0.js. Later we determine how we want to
+      // report and block PRs based on bundle sizes of other dist files.
+      const baseRuntimeBundleSize = masterBundleSizes['dist/v0.js'];
+      const baseRuntimeBundleSizeDelta =
+        prBundleSizes['dist/v0.js'] - baseRuntimeBundleSize;
 
       await db('checks')
-        .update({delta: bundleSizeDelta})
+        .update({delta: baseRuntimeBundleSizeDelta})
         .where({head_sha: check.head_sha});
 
       const requiresApproval =
-        bundleSizeDelta > process.env['MAX_ALLOWED_INCREASE'];
+        baseRuntimeBundleSizeDelta > process.env['MAX_ALLOWED_INCREASE'];
```

Comment:
Suggestion (feel free to ignore):

```javascript
function checkOutput(baseRuntimeBundleSizeDelta, otherBundleSizeDelta, missingBundleSizes) {
  const requiresApproval = baseRuntimeBundleSizeDelta > process.env['MAX_ALLOWED_INCREASE'];
  if (requiresApproval) {
    return { conclusion: 'action_required', output: failedCheckOutput(...) };
  } else {
    return { conclusion: 'success', output: successCheckOutput(...) };
}

.....

Object.assign(updatedCheckOptions, checkOutput(...))
```

Reply:
Punting this for now, though this entire file could use a refactor at some point